src/resmom/stage_func.c

```diff
@@ -1650,16 +1650,15 @@ sys_copy(int dir, int rmtflg, char *owner, char *src, struct rqfpair *pair, int
 		}
 
 		/*
-		 ** There is a problem where scp can return zero
-		 ** indicating success even though the copy did
-		 ** not work.  In the case of a remote stagein,
-		 ** save the mtime of an existing file so we can
-		 ** check the mtime after the copy to be sure it
-		 ** worked.
+		 * There is a problem where scp can return zero indicating success even
+		 * though the copy did not work.  In the case of a remote stagein,
+		 * where the file already exists, save the ctime of an existing file so
+		 * we can check the ctime after the copy to be sure it worked.  ctime
```

Comment:
I am a little confused here, scp is issued with "-p" options which should ideally preserve the m_time from the original file. Why are we switching to c_time?

Reply:
For exactly that reason.  If the same file is copied with `scp -p`, the mtime will not change and thus the original code would not detect any change even if the file transfer were successful.  The test_staging test does exact that; stage in a file that had already exist and preserves the mtime.  As I noted in the comments, ctime changes anytime the file metadata is changed, which includes changing the mtime.